src/display/editor/tools.js

```diff
@@ -377,8 +405,54 @@ class AnnotationEditorUIManager {
 
   #previousActiveEditor = null;
 
+  #boundOnEditingAction = this.onEditingAction.bind(this);
+
+  #previousStates = {
+    isEditing: false,
+    isEmpty: true,
+    hasEmptyClipboard: true,
+    hasSomethingToUndo: false,
+    hasSomethingToRedo: false,
+    hasSelectedEditor: false,
+  };
+
   constructor(eventBus) {
     this.#eventBus = eventBus;
+    this.#eventBus._on("editingaction", this.#boundOnEditingAction);
```

Comment:
It doesn't seem that anything is ever removing this listener, as far as I can tell?
That means that when multiple documents have been opened in the GENERIC viewer, there'll be a number of old (and stale) listeners still around. While "editingaction"-support will *currently* only be implemented in Firefox, this is nonetheless something that we should fix here to prevent any future issues.

Basically, I think that the simplest solution would be to add e.g. a `destroy`-method to this class and having it run:
```javascript
this.#eventBus._off("editingaction", this.#boundOnEditingAction);
```
Then, in the viewer, remove line https://github.com/mozilla/pdf.js/blob/a1ac1a61b75064ef274f69da4aa0da7be9a575dd/web/base_viewer.js#L902
and instead add
```javascript
if (this.#annotationEditorUIManager) {
  this.#annotationEditorUIManager.destroy();
  this.#annotationEditorUIManager = null;
}
```
just after line https://github.com/mozilla/pdf.js/blob/a1ac1a61b75064ef274f69da4aa0da7be9a575dd/web/base_viewer.js#L642